### DISCUSSION TITLE
Ask more follow up questions

### DISCUSSION THREAD

**Discussion by M-Gonzalo** [cline/cline#806] [Feature Requests]
It looks like Cline assumes (on purpose or not) that it's working on a brand new project, or at least it tries to write everything itself, in the same file we're working on. But on a real world project, there are tons of "there's a thing for that" facilities. Be it schemas, types, services, etc. I usually try to tell it where to find anything it might need but it would be better to straight up ask something like "do we have an utility for this, or do you want me to write it now?"

**Comment by RadikXMi** [comment]
> It looks like Cline assumes (on purpose or not) that it's working on a brand new project, or at least it tries to write everything itself, in the same file we're working on. But on a real world project, there are tons of "there's a thing for that" facilities. Be it schemas, types, services, etc. I usually try to tell it where to find anything it might need but it would be better to straight up ask something like "do we have an utility for this, or do you want me to write it now?"
